web/assets/js/GoogleAnalyticsService.js

```diff
@@ -0,0 +1,77 @@
+class GoogleAnalyticsService {
+
+    constructor() {
+        this.gaDataAttr = 'data-ga-action';
+        this.gaCategoryAttr = 'data-ga-category';
+        this.gaLabelAttr = 'data-ga-label';
+    }
+
+    init() {
+        const gaSelectors = document.querySelectorAll(`[${this.gaDataAttr}]`);
+
+        if (gaSelectors.length > 0) {
+            gaSelectors.forEach(selector => {
+                selector = this.rewriteIfHasChildSelector(selector);
+                selector.addEventListener('click', (event) => {
+                    this.sendGARequest(this.getGAItemObject(event.currentTarget));
+                });
+            });
+        }
+    }
+
+    rewriteIfHasChildSelector(selector) {
+        if (Object.values(selector.classList).indexOf('ga-link-listener') >= 0) {
+            const childSelector = selector.querySelector('a');
+            selector = this.rewriteDataAttributes(selector, childSelector);
+        }
+
+        return selector;
+    }
+
+    getGAItemObject(selector) {
+        return new GoogleAnalyticsItem(
```

Comment:
Just return a new object instead of class instance, for better performance and code readability.